### PR TITLE
Add script/setup

### PR DESCRIPTION
**Pull request by bkeepers**
This is an attempt to fix the TODO in the README for adding `script/setup`. This works on my machine, but there's still some cleanup:

- [x] Update docs in README
- [x] Detect if homebrew is installed
- [x] Make idempotent

### CODE REVIEW

File: script/setup [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+open /Applications/Hammerspoon.app
+open /Applications/Karabiner-Elements.app
+
+# Enable apps at startup
+osascript -e 'tell application "System Events" to make login item at end with properties {path:"/Applications/Hammerspoon.app", hidden:true}'
+osascript -e 'tell application "System Events" to make login item at end with properties {path:"/Applications/Karabiner-Elements.app", hidden:true}'
+
+# Enable accessibility - https://github.com/jacobsalmela/tccutil
+# FIXME Doesn't work with SIP https://github.com/jacobsalmela/tccutil/issues/18
+# sudo tccutil -e org.hammerspoon.Hammerspoon
```
Comment (bkeepers) [inline script/setup]: @mikemcquaid do you know any tricks to programmatically enable Accessibility?

Comment (MikeMcQuaid) [inline script/setup]: I do not, sorry.

File: Brewfile [hunk 2]
```diff
@@ -0,0 +1,6 @@
+cask_args appdir: '/Applications'
```
Comment (MikeMcQuaid) [inline Brewfile]: This is the default now so can be dropped.

File: script/setup [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+# Prepare custom settings for Karabiner-Elements
+mkdir -p ~/.config/karabiner/
+ln -sf $PWD/karabiner/karabiner.json ~/.config/karabiner/karabiner.json
+
+# Prepare custom settings for Hammerspoon
+ln -sf $PWD/hammerspoon ~/.hammerspoon
+
+# Prepare custom settings for navigating between words in iTerm2
+cat $PWD/inputrc >> ~/.inputrc
```
Comment (MikeMcQuaid) [inline script/setup]: Pedantry: may want to check the contents aren't already present to avoid appending multiple times.

Comment (bkeepers) [inline script/setup]: 👍 done in be820cf

File: script/setup (hunk 1 above)
Comment (bkeepers) [inline script/setup]: No problem, thanks.

File: script/setup [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+# Prepare custom settings for Karabiner-Elements
+mkdir -p ~/.config/karabiner/
+ln -sf $PWD/karabiner/karabiner.json ~/.config/karabiner/karabiner.json
+
+# Prepare custom settings for Hammerspoon
+ln -sfn $PWD/hammerspoon ~/.hammerspoon
+
+# Prepare custom settings for navigating between words in iTerm2
+grep -q forward-word ~/.inputrc || cat $PWD/inputrc >> ~/.inputrc
```
Comment (bkeepers) [inline script/setup]: 👍 Fixed by adding `-s` flag:

```
     -s, --no-messages
             Silent mode.  Nonexistent and unreadable files are ignored (i.e.
             their error messages are suppressed).
```

**Comment by jasonrudolph** [comment]
Thanks, @bkeepers! 🍻

**Pull request merged**